src/Optimizer/Transformer/AmpStoryCssOptimizer.php

```diff
@@ -0,0 +1,244 @@
+
+        foreach ($document->head->childNodes as $childNode) {
+            if (! $childNode instanceof Element) {
+                continue;
+            }
+
+            if ($this->isAmpStoryScript($childNode)) {
+                $hasAmpStoryScript = true;
+                continue;
+            }
+
+            if ($this->isAmpStoryDvhPolyfillScript($childNode)) {
+                $hasAmpStoryDvhPolyfillScript = true;
+                continue;
+            }
+
+            if ($this->isStyleAmpCustom($childNode)) {
+                $styleAmpCustom = $childNode;
+                continue;
+            }
+        }
+
+        // We can return early if no amp-story script is found.
+        if (! $hasAmpStoryScript) {
+            return;
+        }
+
+        $this->appendAmpStoryCssLink($document);
+
+        if ($styleAmpCustom) {
+            $this->modifyAmpCustomCSS($styleAmpCustom);
+            // Make sure to not double install the dvh polyfill.
+            if (! $hasAmpStoryDvhPolyfillScript) {
+                $this->appendAmpStoryDvhPolyfillScript($document);
+            }
+        }
+
+        $this->supportsLandscapeSSR($document);
+        $this->aspectRatioSSR($document);
+    }
+
+    /**
+     * Check whether the element is an AMP Story element.
+     *
+     * @param Element $element Element to check.
+     * @return bool Whether the given element is an AMP story.
+     */
+    private function isAmpStoryScript(Element $element)
+    {
+        return ($element->tagName === Tag::SCRIPT)
+            && ($element->getAttribute(Attribute::CUSTOM_ELEMENT) === Extension::STORY);
+    }
+
+    /**
+     * Check whether the element is a script[amp-story-dvh-polyfill] element.
+     *
+     * @param Element $element Element to check.
+     * @return bool Whether the element is a script[amp-story-dvh-polyfill] element.
+     */
+    private function isAmpStoryDvhPolyfillScript(Element $element)
+    {
+        return $element->tagName === Tag::SCRIPT && $element->hasAttribute(Attribute::AMP_STORY_DVH_POLLYFILL);
+    }
+
+    /**
+     * Check whether the element is a style[amp-custom] element.
+     *
+     * @param Element $element Element to check.
+     * @return bool Whether the element is a style[amp-custom] element.
+     */
+    private function isStyleAmpCustom(Element $element)
+    {
+        return $element->tagName === Tag::STYLE && $element->hasAttribute(Attribute::AMP_CUSTOM);
+    }
+
+    /**
+     * Insert a link element with amp-story css source.
+     *
+     * @param Document $document Document to append the link.
+     */
+    private function appendAmpStoryCssLink(Document $document)
+    {
+        $ampStoryCssLink = $document->createElementWithAttributes(Tag::LINK, [
+            Attribute::REL           => Attribute::REL_STYLESHEET,
+            Attribute::AMP_EXTENSION => Extension::STORY,
+            Attribute::HREF          => Amp::CACHE_HOST . '/v0/amp-story-1.0.css',
```

Comment:
This has a version hard-coded here.

Please add a `TODO` that we need to take the following into account when deciding on a version:
- latest stable version available
- channel that the runtime is locked to, i.e. whether LTS is active

Reply:
Added the TODO.